### PR TITLE
Add progress bar keywords

### LINKED ISSUE #3190: Allow specification of ProgressBar keyword arguments in the AnalysisBase run method
The ProgressBar (subclass of tqdm) in the AnalysisBase doesn't currently take in keyword arguments nor does it give much access to the underlying functionality of tqdm. In particular, the position keyword argument cannot be changed to adjust the location of the bar on screen. When spreading analyses across multiple processes, the rendered bars are constantly overwriting each other.

https://github.com/MDAnalysis/mdanalysis/blob/6282385644d672cf7023584347e35048935ab429/package/MDAnalysis/analysis/base.py#L186

### PR DESCRIPTION
**Pull request by marinegor**
Fixes #3190 

Changes made in this Pull Request:
 - add keyword-only `pbar_kwargs` argument to the `AnalysisBase.run` method

This presumably fixes the comments raised in [this](https://github.com/MDAnalysis/mdanalysis/pull/3191) PR which was closed earlier, since the keyword-only mechanism prevents capturing other arguments during dictionary unpacking.

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

----
:books: Documentation preview :books:: https://readthedocs-preview--4085.org.readthedocs.build/en/4085/

### CODE REVIEW

**Comment by marinegor** [comment]
Hi everyone, I'm not entirely sure what to do with the failing tests -- afais all of them fail with:

`Error: There was an error fetching the storage URL during POST: 500 - Unknown error, please try again later`

error during codecov report upload, so presumably issue is not with my code. Locally all tests run successfully.

**Comment by hmacdope** [comment]
@marinegor yes not your fault, codecov is super flaky. Please fix conflicts and then ill review :)

File: package/MDAnalysis/analysis/base.py [hunk 1]
```diff
@@ -411,6 +411,10 @@ def run(self, start=None, stop=None, step=None, frames=None,
 
         verbose : bool, optional
             Turn on verbosity
+        
+        pbar_kwargs : dict, optional
```
Comment (yuxuanzhuang) [inline package/MDAnalysis/analysis/base.py]: please added a versionadded mark (2.5.0) to it and a versionchanged below.

File: testsuite/MDAnalysisTests/analysis/test_base.py [hunk 2]
```diff
@@ -30,7 +30,7 @@
 from numpy.testing import assert_equal, assert_almost_equal
 
 import MDAnalysis as mda
-from MDAnalysis.analysis import base
+from MDAnalysis.analysis import base, rms
```
Comment (yuxuanzhuang) [inline testsuite/MDAnalysisTests/analysis/test_base.py]: I assume you don't really need this?

File: testsuite/MDAnalysisTests/analysis/test_base.py [hunk 3]
```diff
@@ -274,6 +274,12 @@ def test_verbose_progressbar_run(u, capsys):
     actual = err.strip().split('\r')[-1]
     assert actual[:24] == expected[:24]
 
+def test_verbose_progressbar_run_with_kwargs(u, capsys):
+    an = FrameAnalysis(u.trajectory).run(verbose=True, pbar_kwargs={'desc':'custom'})
+    out, err = capsys.readouterr()
+    expected = u'custom: 100%|██████████| 98/98 [00:00<00:00, 8799.49it/s]'
+    actual = err.strip().split('\r')[-1]
+    assert actual[:24] == expected[:24]
```
Comment (yuxuanzhuang) [inline testsuite/MDAnalysisTests/analysis/test_base.py]: this is just nitpicking but 24 is for inclusion of text until `98/98` in the previous text. You could update this number accordingly.

File: package/MDAnalysis/analysis/base.py (hunk 1 above)
Comment (yuxuanzhuang) [reply package/MDAnalysis/analysis/base.py]: Also, add the change to the CHANGELOG file.

**Comment by marinegor** [comment]
Hey @hmacdope I wonder what would be the best way to confidently resolve the "package/AUTHORS" merge conflict -- I pulled upstream & resolved conflict, but since new contributions keep coming to the `develop`, it keeps happening🤔

**Comment by hmacdope** [comment]
You will have to update your PR every time develop changes, unfortunately thats just a byproduct of how distributed code development works.

**Comment by marinegor** [comment]
Ah, it's fine then!
Could you please review the PR then regardless of the merge conflicts, which happen to be only in the AUTHORS file?

File: package/MDAnalysis/analysis/base.py [hunk 4]
```diff
@@ -390,7 +390,7 @@ def _conclude(self):
         pass  # pylint: disable=unnecessary-pass
 
     def run(self, start=None, stop=None, step=None, frames=None,
-            verbose=None):
+            verbose=None, *, pbar_kwargs={}):
```
Comment (hmacdope) [inline package/MDAnalysis/analysis/base.py]: I would like the name to be `progressbar_kwargs` to be slightly clearer.

File: package/MDAnalysis/analysis/base.py (hunk 1 above)
Comment (hmacdope) [inline package/MDAnalysis/analysis/base.py]: Same change here.

**Review by hmacdope — CHANGES_REQUESTED** [review]
One quick change, but other than that looks good. You will need to add a `CHANGELOG` entry also and introduce yourself on the developer/gsoc mailing list if you havnt already.

File: package/MDAnalysis/analysis/base.py (hunk 4 above)
Comment (marinegor) [reply package/MDAnalysis/analysis/base.py]: Fixed!

File: package/MDAnalysis/analysis/base.py (hunk 1 above)
Comment (marinegor) [reply package/MDAnalysis/analysis/base.py]: Fixed too.

**Review by yuxuanzhuang — CHANGES_REQUESTED** [review]

**Comment by yuxuanzhuang** [comment]
Sorry if a few comments are outdated already. I didn't realize I need to submit my review. I thought "pending" means pending for feedback :⁠-⁠)

File: testsuite/MDAnalysisTests/analysis/test_base.py (hunk 3 above)
Comment (marinegor) [reply testsuite/MDAnalysisTests/analysis/test_base.py]: Updated! Thanks, I didn't get the meaning of this number from context :)

File: testsuite/MDAnalysisTests/analysis/test_base.py (hunk 2 above)
Comment (marinegor) [reply testsuite/MDAnalysisTests/analysis/test_base.py]: Indeed, removed.

File: package/MDAnalysis/analysis/base.py (hunk 1 above)
Comment (marinegor) [reply package/MDAnalysis/analysis/base.py]: Updated CHANGELOG and added `versionadded` mark.

**Comment by marinegor** [comment]
I apologize for the little mess with requesting code -- hope you didn't get much annoying notifications about it :)

**Review by yuxuanzhuang — APPROVED** [review]
LGTM!

**Review by hmacdope — APPROVED** [review]
LGTM

**Pull request merged**